contracts/vaults/VoltzVault.sol

```diff
@@ -278,167 +124,201 @@ contract VoltzVault is IVoltzVault, IntegrationVault {
         returns (uint256[] memory actualTokenAmounts)
     {
         actualTokenAmounts = new uint256[](1);
-        if (_currentPosition.low == 0 && _currentPosition.high == 0) {
-            return actualTokenAmounts;
-        }
+        actualTokenAmounts[0] = tokenAmounts[0];
+        _updateMargin(tokenAmounts[0].toInt256());
+        _updateLiquidity(tokenAmounts[0].toInt256() * _leverage.toInt256());
+    }
 
-        Options memory opts = _parseOptions(options);
-        if (opts.updatePosition) {
-            _updatePosition(TickRange(opts.newTickLow, opts.newTickHigh));
+    function _pull(
+        address to,
+        uint256[] memory tokenAmounts,
+        bytes memory options
+    ) internal override returns (uint256[] memory actualTokenAmounts) {
+        actualTokenAmounts = new uint256[](1);
+        uint256 termEndTimestampWad = _marginEngine.termEndTimestampWad();
+         if (termEndTimestampWad > Time.blockTimestampScaled()) {
+            return actualTokenAmounts;
         }
```

Comment:
here we need to check if the positions are already settled. Otherwise, only the first user can withdraw and then all the other withdrawals will be reverted with `Position already settled`.